### PR TITLE
3.x dev cache

### PR DESCRIPTION
**Pull request by dugenkui03**
## What is the purpose of the change

improve the implementation of cache，especially ThreadLocalCache and ExpiringCache, to prevents memory leak.

## Brief changelog

XXXXX

## Verifying this change

XXXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

- [x] Make sure there is a [GITHUB_issue](https://github.com/apache/incubator-dubbo/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a GITHUB issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Format the pull request title like `[Dubbo-XXX] Fix UnknownException when host config not exist #XXX`. Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/incubator-dubbo/tree/master/dubbo-test).
- [ ] Run `mvn clean install -DskipTests` & `mvn clean test-compile failsafe:integration-test` to make sure unit-test and integration-test pass.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/incubator-dubbo/wiki/Software-donation-guide).

### CODE REVIEW

File: dubbo-filter/dubbo-filter-cache/src/main/java/org/apache/dubbo/cache/Cache.java [hunk 1]
```diff
@@ -25,4 +25,10 @@
 
     Object get(Object key);
 
+    boolean remove(Object key);
```
Comment (kimmking) [inline dubbo-filter/dubbo-filter-cache/src/main/java/org/apache/dubbo/cache/Cache.java]: how to use it?

Comment (dugenkui03) [reply dubbo-filter/dubbo-filter-cache/src/main/java/org/apache/dubbo/cache/Cache.java]: when user want to Invalidate cache,espescially in ThreadLocalCache.details about this commit can be found in [#2592](https://github.com/apache/incubator-dubbo/issues/2592)

**Comment by carryxyh** [comment]
Do not agree with this pr. Unless a complete set of expiration policies can be provided.

**Comment by carryxyh** [comment]
Hi, @dugenkui03 
I will close this pr. Expect you to be able to provide a complete cleanup caching strategy.
:)

**Pull request closed without merging**